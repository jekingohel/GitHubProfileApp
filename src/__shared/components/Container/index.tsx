import React from 'react';
import {ViewStyle} from 'react-native';
import View from '../View';

interface CustomViewProps {
  children: React.ReactNode;
  style?: ViewStyle;
}

const Container: React.FC<CustomViewProps> = props => {
  return (
    <View
      style={{
        backgroundColor: '#FFFFFF',
        paddingLeft: 20,
        paddingRight: 20,
        paddingTop: 20,
        paddingBottom: 20,
      }}>
      {props.children}
    </View>
  );
};

export default Container;
