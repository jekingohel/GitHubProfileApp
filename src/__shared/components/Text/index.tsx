import React from 'react';
import {Text as RNText, TextStyle} from 'react-native';

type FontWeight =
  | 'normal'
  | 'bold'
  | '100'
  | '200'
  | '300'
  | '400'
  | '500'
  | '600'
  | '700'
  | '800'
  | '900'
  | undefined;

type Align = 'left' | 'auto' | 'right' | 'center' | 'justify' | undefined;

interface CustomViewProps {
  children: React.ReactNode;
  fontSize?: number;
  fontWeight?: FontWeight;
  align?: Align;
  color?: string;
  italic?: boolean;
}

const Text: React.FC<CustomViewProps> = props => {
  const {
    fontSize = 12,
    fontWeight = '400',
    align = 'left',
    color = 'black',
    italic = false,
  } = props;

  const containerStyle: TextStyle = {
    fontSize,
    fontWeight,
    textAlign: align,
    color,
    fontStyle: italic ? 'italic' : 'normal',
  };

  return <RNText style={containerStyle}>{props.children}</RNText>;
};

export default Text;
