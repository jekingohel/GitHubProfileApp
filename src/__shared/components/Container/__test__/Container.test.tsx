import React from 'react';
import {render} from '@testing-library/react-native';
import Container from '..';
import Text from '../../Text';

test('renders Container with default styles', () => {
  const {toJSON} = render(
    <Container>
      <Text>Test Content</Text>
    </Container>,
  );
  expect(toJSON()).toMatchSnapshot();
});

test('renders Container with custom styles', () => {
  const customStyles = {
    backgroundColor: '#EFEFEF',
    paddingLeft: 10,
    paddingRight: 10,
    paddingTop: 15,
    paddingBottom: 15,
  };

  const {toJSON} = render(
    <Container style={customStyles}>
      <Text>Test Content</Text>
    </Container>,
  );
  expect(toJSON()).toMatchSnapshot();
});

test('renders Container with children', () => {
  const {getByText, toJSON} = render(
    <Container>
      <Text>Test Content</Text>
    </Container>,
  );

  const childText = getByText('Test Content');
  expect(childText).toBeTruthy();
  expect(toJSON()).toMatchSnapshot();
});
