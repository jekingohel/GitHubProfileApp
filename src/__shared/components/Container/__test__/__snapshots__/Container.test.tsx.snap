// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Container with children 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 12,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 18,
        "textAlign": "left",
      }
    }
  >
    Test Content
  </Text>
</View>
`;

exports[`renders Container with custom styles 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 12,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 18,
        "textAlign": "left",
      }
    }
  >
    Test Content
  </Text>
</View>
`;

exports[`renders Container with default styles 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 12,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 18,
        "textAlign": "left",
      }
    }
  >
    Test Content
  </Text>
</View>
`;
